src/scripts/main.js

```diff
@@ -1 +1,68 @@
 'use strict';
+
+function firstPromise() {
+  return new Promise((resolve, reject) => {
+    document.body.addEventListener('click', () => {
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      reject(new Error('First promise was rejected'));
+    }, 3000);
+  });
+}
+
+function secondPromise() {
+  return new Promise((resolve) => {
+    document.body.addEventListener('click', () => {
+      resolve('Second promise was resolved');
+    });
+
+    document.body.addEventListener('contextmenu', (e) => {
+      e.preventDefault();
+      resolve('Second promise was resolved');
+    });
+  });
+}
+
+function thirdPromise() {
+  return new Promise((resolve) => {
+    let leftClicked = false;
+    let rightClicked = false;
+
+    const checkClicks = () => {
+      if (leftClicked && rightClicked) {
+        resolve('Third promise was resolved');
+      }
+    };
+
+    document.body.addEventListener('click', () => {
+      leftClicked = true;
+      checkClicks();
+    });
+
+    document.body.addEventListener('contextmenu', (e) => {
+      e.preventDefault();
+      rightClicked = true;
+      checkClicks();
+    });
```

Comment:
Similar to 'secondPromise', the 'thirdPromise' function could potentially resolve multiple times if the user clicks multiple times after both flags are set. Ensure that the promise resolves only once by adding a mechanism to prevent multiple resolutions.